src/main/resources/avro/common.avdl

```diff
@@ -387,19 +300,59 @@ record Position {
 }
 
 /**
-A `Segment` is a range on a sequence, possibly including the joins at the
-sequence's ends. It does not include base data. (The bases for a sequence are
-available through the `getSequenceBases()` API call.)
+A `Join` is simply a pair of Position objects, which we call the sides. The are
+logically unordered (i.e. swapping makes no difference), but we require a rank
+on the Sequences, and so implicitly on the positions, so to avoid ambiguity we
+require that the position for side1 is less than that for side2.
```

Comment:
possibly include this in documentation on `Position`, but from what it says here, I'm guessing that it's something like:
```
{(p1, o1), seq1} < {(p2, o2), seq2}  <=>  (seq 1 < seq2) or (seq1 = seq2. and p1 < p2)
```